cli/cmd/summary.go

```diff
@@ -62,3 +73,109 @@ func checkFilePath(filename string) error {
 	dirPath := filepath.Dir(filename)
 	return os.MkdirAll(dirPath, 0755)
 }
+
+func generateSummaryTable() (*schema.Table, error) {
+	tableName := "cloudquery_sync_summaries"
+	t := schema.Tables{{
+		Name: tableName,
+		Transform: transformers.TransformWithStruct(
+			&syncSummary{},
+		),
+	}}
+	if err := transformers.TransformTables(t); err != nil {
+		return nil, err
+	}
+	return t[0], nil
+}
+
+func migrateSummaryTable(writeClient plugin.Plugin_WriteClient, destTransformer *transformer.RecordTransformer, spec specs.Destination) error {
+	if !spec.SyncSummary {
+		return nil
+	}
+
+	summaryTable, err := generateSummaryTable()
+	if err != nil {
+		return err
+	}
+	summaryTableSchema := summaryTable.ToArrowSchema()
+	transformedSchema := destTransformer.TransformSchema(summaryTableSchema)
+	transformedSchemaBytes, err := plugin.SchemaToBytes(transformedSchema)
+	if err != nil {
+		return err
+	}
+	wr := &plugin.Write_Request{}
+	wr.Message = &plugin.Write_Request_MigrateTable{
+		MigrateTable: &plugin.Write_MessageMigrateTable{
+			MigrateForce: spec.MigrateMode == specs.MigrateModeForced,
+			Table:        transformedSchemaBytes,
+		},
+	}
+	if err := writeClient.Send(wr); err != nil {
+		return handleSendError(err, writeClient, "migrate sync summary table")
+	}
+	return nil
+}
+
+func sendSummary(writeClients []plugin.Plugin_WriteClient, destinationSpecs []specs.Destination, destinationsClients []*managedplugin.Client, destinationTransformers []*transformer.RecordTransformer, summary *syncSummary, noMigrate bool) error {
+	summaryTable, err := generateSummaryTable()
+	if err != nil {
+		return err
+	}
+	defaultCaser := caser.New()
+	for i := range destinationsClients {
+		if !destinationSpecs[i].SyncSummary {
+			continue
+		}
+
+		// Only send the summary to the destination that matches the current destination
+		if destinationSpecs[i].Name != summary.DestinationName || destinationSpecs[i].Version != summary.DestinationVersion || destinationSpecs[i].Path != summary.DestinationPath {
```

Comment:
This fix works, but maybe we can drop the looping over destinations and pass a single destination instead?